### PR TITLE
Show in the UI how long the user has been streaming for.

### PR DESCRIPTION
**Pull request by gabek**

### CODE REVIEW

File: webroot/js/app.js [hunk 1]
```diff
@@ -223,8 +223,26 @@ class Owncast {
     }
 
     this.streamStatus = status;
+    this.setCurrentStreamDuration();
```
Comment (gingervitis) [inline webroot/js/app.js:226]: I would just put this call in this.handleOnlineMode().

File: webroot/js/app.js [hunk 2]
```diff
@@ -223,8 +223,26 @@ class Owncast {
...
   };
   
+  setCurrentStreamDuration() {
+    // If we're offline then don't update any of the UI.
+    if (!this.streamStatus.online) {
```
Comment (gingervitis) [inline webroot/js/app.js:231]: If we only do this in handleOnlineMode, we won't need this here.

File: webroot/js/app.js [hunk 3]
```diff
@@ -252,6 +270,8 @@ class Owncast {
     clearTimeout(this.disableChatTimer);
     this.disableChatTimer = null;
     this.messagingInterface.enableChat();
+
+    setInterval(this.setCurrentStreamDuration.bind(this), 1000);
```
Comment (gingervitis) [inline webroot/js/app.js:274]: We should create a new timer instance up top in the constructor for this interval counter., with all the other Timers.

We should make a `this.streamDurationTimer = null` in the contructor. 
Then here `this.streamDurationTimer = setInterval...`;

Comment (gingervitis) [reply webroot/js/app.js:274]: We'll have to make sure that in `handleOfflineMode()` we `clearInterval(this.streamDurationTimer)` .

**Comment by gabek** [comment]
@gingervitis You told me to leave this and let you do it, so that's what I'm doing :P

**Pull request merged**